whole-tale/ngx-dashboard
# Tale versioning panel

## Problem
As a user of the WholeTale platform, it would be nice to be able to save a particular version/state of my Tale to rerun later. This would enable me to save my favorite / working version of the code to preserve existing evaluation, while continuing to evolve the code in new directions.
  
## Approach
Add a panel to the Run view that allows the user to save a new version of the Tale. This panel will list all existing versions/runs created, and offer functionality to create new versions and runs of different configurations.

## How to Test
Prerequisites: at least one Tale created, newest `deploy-dev` + `wt_versioning` (assumes PRs have been merged)

1. Checkout this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Create a Tale, if you haven't already
4. Navigate to the Tale's Run view
    * You should see a new "history" icon on the tab bar at the far right
5. Click on the history icon
    * You should see a timeline appear listing out existing versions for this Tale
    * You should see a button allowing you to "Save Tale Version"
    * [HIDDEN] You should see a dropdown allowing you to choose a Run Configuration
    * [HIDDEN] "Default" should be selected the default selection for Run Configuration  
    * [HIDDEN] You should see a button allowing you to perform a "Recorded Run"
6. [HIDDEN] Expand the "Run Configurations" dropdown
    * You should see a list of existing run configuration
    * You should see a button allowing you to "Edit Run Configurations"
    * TODO: Edit currently does nothing
    * TODO: Wire to API (currently hard-coded / mock data)
7. Click the vertical ellipses beside one of the listed versions
    * You should see a dropdown appear
    * The dropdown should offer options to View Info / Rename / Delete
    * [HIDDEN] The dropdown should offer options to Restore / Compare
    * TODO: Restore / Compare do not yet have API endpoints
8. To test functionality: Save a new Tale version, rename it, view it's info, then delete it
